src/invidious/helpers/youtube_api.cr

```diff
@@ -194,7 +195,10 @@ def _youtube_api_post_json(endpoint, data, client_name : String | Nil = "WEB",
     LOGGER.debug("Sending this data without proxy\n: #{data.to_s}")
     response = YT_POOL.client &.post(
       "#{endpoint}?key=#{hardcoded_api_key}",
-      headers: HTTP::Headers{"content-type" => "application/json; charset=UTF-8"},
+      headers: HTTP::Headers{
+        "content-type"  => "application/json; charset=UTF-8",
+        "Accept-Encoding": "gzip",
```

Comment:
No idea if this is really worth it? It seems to reduce the latency when requesting data from innertube API.

Reply:
Yeah, this is totally worth it :) Compressing the data means that you have less data flowing in the pipes of the internet, so obviously, it takes less time to transmit. In exchange, it costs a little bit more CPU because the data has to be decompressed.

That's something I had in mind for quite some time, but I wasn't sure if it required some more code to decompress, or if the underlying HTTP library does that "automagically".

One thing though: please avoid mixing `Hash` (`key => value`) and NamedTuple  (`key : value`) declarations!